### PR TITLE
Add common Mono Mac paths for echoes exporting

### LINKED ISSUE #4474: Prime 2 Exporter: Expose Mono paths for macOS
Most users start randovania on macOS by double clicking the program from their file explorer. This has the unfortunate disadvantage that RDV is spawned from a non-login non-interactive shell, and thus is not able to find the users Mono path, because those are prepended in login start scripts.
Thus I propose to whenever start mono, prepend the following paths:
- `/Library/Frameworks/Mono.framework/Versions/Current/Commands` = default installation path from mono's website
- `/usr/local/bin` = installation path for programs installed via homebrew on x64 based macs
- `/opt/homebrew/bin` = installation path for programs installed via homebrew for arm based macs.

A user *could* theoretically have installed mono somewhere else, but these three paths should cover 99% of all installation cases.

### PR DESCRIPTION
**Pull request by Miepee**
Fixes #4474

### CODE REVIEW

File: test/games/patchers/test_csharp_subprocess.py [hunk 1]
```diff
@@ -35,8 +42,13 @@ def test_is_windows(mocker, system_name, expected):
 
 
 @pytest.mark.parametrize("add_mono", [False, True])
-def test_process_command_no_thread(mock_is_windows, mocker: pytest_mock.MockerFixture, monkeypatch,
+@pytest.mark.skipif(mock_is_windows and mock_is_mac,
```
Comment (henriquegemignani) [inline test/games/patchers/test_csharp_subprocess.py:45]: Does this actually work?

Comment (Miepee) [reply test/games/patchers/test_csharp_subprocess.py:45]: Double checked, it doesn't. Whoops.
That currently causes it to skip the test in its entirety.

Comment (henriquegemignani) [reply test/games/patchers/test_csharp_subprocess.py]: just do a `if a and b: pytest.skip()` in the function body

Comment (Miepee) [reply test/games/patchers/test_csharp_subprocess.py]: Addressed now
grafik

**Review by henriquegemignani — CHANGES_REQUESTED** [review]
Can you solve the mypy errors too?

File: randovania/games/prime2/patcher/csharp_subprocess.py [hunk 2]
```diff
@@ -40,11 +46,13 @@ async def _read_data(stream: StreamReader, read_callback: Callable[[str], None])
             break
 
 
-async def _process_command_async(args: list[str], input_data: str, read_callback: Callable[[str], None]):
+async def _process_command_async(args: list[str], input_data: str, read_callback: Callable[[str], None],
+                                 envs: _Environ | dict = os.environ):
```
Comment (henriquegemignani) [inline randovania/games/prime2/patcher/csharp_subprocess.py:50]: Change this to be a `addition_path_entries: collections.abc.Sequence[str]`, then make this function responsible for creating the new env dict.

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

**Review by henriquegemignani — APPROVED** [review]

**Pull request merged**